File: backend/app/api/jobs.py
from fastapi import APIRouter, Query

from app.schemas.job import Job
from app.services.jobs_service import jobs_service


router = APIRouter(
    prefix="/api/jobs",
    tags=["Jobs"],
)


@router.get(
    "/expiring",
    response_model=list[Job],
)
async def get_expiring_jobs(
    days: int = Query(
        default=7,
        ge=0,
        le=365,
    ),
):
    return await jobs_service.get_expiring_jobs(
        days=days
    )

@router.get(
    "/overdue",
    response_model=list[Job],
)
async def get_overdue_jobs():
    return await jobs_service.get_overdue_jobs()

File: frontend/src/api/jobs.ts
import type {
  Job,
} from "../types/job"


const API_BASE_URL =
  "http://localhost:8000"


export async function getOpenJobs():
Promise<Job[]> {

  const response =
    await fetch(
      `${API_BASE_URL}/api/jobs/open`
    )


  if (!response.ok) {
    throw new Error(
      "Error loading open jobs"
    )
  }


  return response.json()
}


export async function getExpiringJobs(
  days = 7
): Promise<Job[]> {

  const response =
    await fetch(
      `${API_BASE_URL}/api/jobs/expiring?days=${days}`
    )


  if (!response.ok) {
    throw new Error(
      "Error loading expiring jobs"
    )
  }


  return response.json()
}


export async function getOverdueJobs():
Promise<Job[]> {

  const response =
    await fetch(
      `${API_BASE_URL}/api/jobs/overdue`
    )


  if (!response.ok) {
    throw new Error(
      "Error loading overdue jobs"
    )
  }


  return response.json()
}